phalcon/Mvc/Model.zep

```diff
@@ -1570,9 +1570,9 @@ abstract class Model extends AbstractInjectionAware implements EntityInterface,
             }
 
             if globals_get("orm.cast_on_hydrate") {
-                let hasChanged = value != snapshot[name];
-            } else {
                 let hasChanged = value !== snapshot[name];
+            } else {
+                let hasChanged = value != snapshot[name];
```

Comment:
This fix conflicts with previous issues: 

https://github.com/phalcon/cphalcon/pull/12655
https://forum.phalcon.io/discussion/15529/bug-regarding-getchangedfields-and-haschanged#C45969

Reply:
Could you please correct the test?

Reply:
The purpose of this test is to check if the `Model` detects the difference between `null` and `0`.
We could keep this because the snapshot data stores `null` correctly with both `cast_on_hydrate` settings.

If either the actual or the snapshot value is `null`, we could use the `!==`

Reply:
https://travis-ci.org/phalcon/cphalcon/jobs/583222050#L601

Reply:
> https://travis-ci.org/phalcon/cphalcon/jobs/583222050#L601

@Jeckerson, yes, I was talking about this test:

```php
$I->assertNotEmpty([]) at tests/integration/Mvc/Model/Refactor-SnapshotCest.php:370
```

This failed because it couldn't tell the difference between `deleted = 0` and `deleted = null` (`orm.cast_on_hydrate` was off)

Reply:
@zsilbi I've discussed this with the team. Can you keep the compare as it is? If you cast your model you should set your properties as the required type otherwise we expect a hasChanged to be true.

Reply:
Thanks @ruudboon. I removed the comparison and the test for it, and rebased.